Tests/SonarScanner.MSBuild.PreProcessor.Test/CacheProcessorTests.cs

```diff
@@ -135,11 +145,112 @@ public void PullRequestCacheBasePath_NoSonarScannerWorkingDirectory_IsNull()
+        {
+            var context = new CacheContext(TestContext, "/k:key-no-cache /d:sonar.pullrequest.base=TARGET_BRANCH");
+            await context.Sut.Execute();
+
+            context.Logger.AssertInfoLogged("Processing pull request with base branch 'TARGET_BRANCH'.");
+            context.Logger.AssertInfoLogged("Cache data is not available. Incremental PR analysis is disabled.");
+            context.Sut.UnchangedFilesPath.Should().BeNull();
+        }
+
+        [TestMethod]
+        public async Task Execute_PullRequest_FullProcessing_WithCache()
+        {
+            var context = new CacheContext(TestContext, "/k:key /d:sonar.pullrequest.base=TARGET_BRANCH");
+            await context.Sut.Execute();
+
+            context.Logger.AssertInfoLogged("Processing pull request with base branch 'TARGET_BRANCH'.");
+            context.Sut.UnchangedFilesPath.Should().EndWith("UnchangedFiles.txt");
+        }
+
+        [TestMethod]
+        public void ProcessPullRequest_EmptyCache_DoesNotProduceOutput()
+        {
+            using var sut = CreateSut();
+            var cache = new AnalysisCacheMsg();
+            sut.ProcessPullRequest(cache);
+
+            sut.UnchangedFilesPath.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void ProcessPullRequest_AllFilesChanged_DoesNotProduceOutput()
+        {
+            using var context = new CacheContext(TestContext);
+            foreach (var path in context.Paths)
+            {
+                File.WriteAllText(path, "Content of this file has changed");
+            }
+            context.ProcessPullRequest();
+
+            context.Sut.UnchangedFilesPath.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void ProcessPullRequest_NoFilesChanged_ProducesAllFiles()
+        {
+            using var context = new CacheContext(TestContext);
+            context.ProcessPullRequest();
+
+            context.Sut.UnchangedFilesPath.Should().EndWith("UnchangedFiles.txt");
+            File.ReadAllLines(context.Sut.UnchangedFilesPath).Should().BeEquivalentTo(context.Paths);
+        }
+
+        [TestMethod]
+        public void ProcessPullRequest_SomeFilesChanged_ProducesOnlyUnchangedFiles()
+        {
+            using var context = new CacheContext(TestContext);
+            CreateFile(context.Root, "AddedFile.cs", "// This file is not in cache", Encoding.UTF8);
+            File.Delete(context.Paths.First());       // This file was in cache, but doesn't exist anymore
+            File.WriteAllText(context.Paths.Last(), " // This file was modified");
+            context.ProcessPullRequest();
+
+            context.Sut.UnchangedFilesPath.Should().EndWith("UnchangedFiles.txt");
+            File.ReadAllLines(context.Sut.UnchangedFilesPath).Should().BeEquivalentTo(new[] { context.Paths[1] });  // Only a single file was not modified
+        }
+
+        private static string Serialize(byte[] value) =>
```

Comment:
We want to have the static methods at the end. Now there is a mix. One static one not, ...

Reply:
I had to shuffle them a bit more to make `CreateProcessedArgs` overloads consecutive